rlog.go

```diff
@@ -209,49 +235,44 @@ func (f filter) match(filename string, level int) (bool, bool) {
-// global variables.
-var settingShowCallerInfo bool = false // whether we log caller info
-var settingDateTimeFormat string       // flags for date/time output
-var settingLogFile string = ""         // logfile name
-
-var logWriterStream *log.Logger       // the first writer to which output is sent
-var logWriterFile *log.Logger         // the second writer to which output is sent
-var logfilterSpec = new(filterSpec)   // filters for log messages
-var tracefilterSpec = new(filterSpec) // filters for trace messages
-
 // init extracts settings for our logger from environment variables when the
-// module is imported.
+// module is imported and calls actual initialization function with that
+// configuration.
 func init() {
-	logLevelEnv := os.Getenv("RLOG_LOG_LEVEL")
-	callerInfoEnv := os.Getenv("RLOG_CALLER_INFO")
-	traceLevelEnv := os.Getenv("RLOG_TRACE_LEVEL")
-	dontLogTimeEnv := os.Getenv("RLOG_LOG_NOTIME")
-	logTimeFormatEnv := os.Getenv("RLOG_TIME_FORMAT")
-	logFileEnv := os.Getenv("RLOG_LOG_FILE")
-	logStreamEnv := strings.ToUpper(os.Getenv("RLOG_LOG_STREAM"))
-
-	// Evaluating the caller info variable.
-	settingShowCallerInfo = isTrueBoolString(callerInfoEnv)
-
-	// Evaluating whether date/time should be logged with each message
-	// By default (if flag is not set) we want to log date and time.
-	logTimeDate := !isTrueBoolString(dontLogTimeEnv)
+	var config rlogConfig = rlogConfig{
+		logLevel:       os.Getenv("RLOG_LOG_LEVEL"),
+		traceLevel:     os.Getenv("RLOG_TRACE_LEVEL"),
+		logTimeFormat:  os.Getenv("RLOG_TIME_FORMAT"),
+		logFile:        os.Getenv("RLOG_LOG_FILE"),
+		logStream:      strings.ToUpper(os.Getenv("RLOG_LOG_STREAM")),
+		logTimeDate:    !isTrueBoolString(os.Getenv("RLOG_LOG_NOTIME")),
+		showCallerInfo: isTrueBoolString(os.Getenv("RLOG_CALLER_INFO")),
+	}
+	Initialize(config)
+}
+
+// Initialize translates config items into initialized data structures,
+// config values and freshly created or opened config files, if necessary.
+// This function prepares everything for the fast and efficient processing of
+// the actual log functions.
+func Initialize(config rlogConfig) {
+	settingShowCallerInfo = config.showCallerInfo
 
 	// Initialize filters for trace (by default no trace output) and log levels
 	// (by default INFO level).
-	tracefilterSpec.fromString(traceLevelEnv, true, noTraceOutput)
-	logfilterSpec.fromString(logLevelEnv, false, levelInfo)
+	logFilterSpec = new(filterSpec)
+	traceFilterSpec = new(filterSpec)
+	traceFilterSpec.fromString(config.traceLevel, true, noTraceOutput)
+	logFilterSpec.fromString(config.logLevel, false, levelInfo)
 
 	// Evaluate the specified date/time format
 	settingDateTimeFormat = ""
-	if logTimeDate {
+	if config.logTimeDate {
 		// Store the format string for date/time logging. Allowed values are
 		// all the constants specified in
 		// https://golang.org/src/time/format.go.
 		var f string
-		switch logTimeFormatEnv {
+		switch config.logTimeFormat {
 		case "ANSIC":
```

Comment:
I can make this comparison here case insensitive, ok. The general format string, however, has to remain case sensitive, since people can define custom formats.